pkg/providers/github.go

```diff
@@ -49,6 +49,14 @@ func filterAssets(as []*github.ReleaseAsset) (*githubFileInfo, error) {
 			matches = append(matches, &githubFileInfo{a.GetBrowserDownloadURL(), a.GetName()})
 		}
 	}
+	if len(matches) == 0 {
```

Comment:
Hmm this is a big dangerous.. since you could potentially be downloading a binary that's not for your OS or Arch..

Did you add this tying the `jq` case? I believe in that case it's not necessary since jq has both arch and os in the filename.. 

Not sure if I like this approach since it's very likely that you could end up downloading the incorrect binary. 

If someone publishes something like `jq-linux` I don't think we should support it since that could be any platform.

Reply:
If only one asset matches either OS or Arch then it's the best bet. If several match _either_ OS _or_ Arch then the user currently gets an ordered list to choose from.

Reply:
> If someone publishes something like `jq-linux` I don't think we should support it since that could be any platform.

PS: Yes that is true, but if it is the only match and it happens to be correct, then isn't it better to err on the side of getting a binary?